software/src/applets/WTVCO.h

```diff
@@ -0,0 +1,691 @@
+                waveform[j] = static_cast<int8_t>(value);
+            }
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_SinusoidalStep(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            waveform[i] = static_cast<int8_t>(127 * std::round(std::sin(TWO_PI * i / size)));
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_WavyStaircase(size_t size) {
+        std::vector<int8_t> waveform(size);
+        size_t stepSize = size / 8;
+        for (size_t i = 0; i < size; ++i) {
+            size_t step = i / stepSize;
+            waveform[i] = static_cast<int8_t>(127 * std::sin(TWO_PI * step / 8));
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_Stepped(size_t size, size_t steps = 8) {
+        std::vector<int8_t> waveform(size);
+        size_t stepSize = size / steps;
+        for (size_t i = 0; i < size; ++i) {
+            waveform[i] = static_cast<int8_t>(127 * ((i / stepSize) / static_cast<double>(steps - 1)) - 128);
+        }
+        return waveform;
+    }
+
+
+
+// noise
+
+    std::vector<int8_t> GenerateWaveForm_Noise(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            waveform[i] = static_cast<int8_t>(rand() % 256 - 128); // Random int8_t values
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_Noise2(size_t size) {
+        std::vector<int8_t> waveform(size);
+        std::random_device rd;
+        std::mt19937 gen(rd());
+        std::uniform_int_distribution<int> dist(-128, 127);
+        for (size_t i = 0; i < size; ++i) {
+            waveform[i] = static_cast<int8_t>(dist(gen));
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_GaussianNoise(size_t size) {
+        std::vector<int8_t> waveform(size);
+        std::random_device rd;
+        std::mt19937 gen(rd());
+        std::normal_distribution<> dist(0, 50); // Mean 0, std deviation 50
+        for (size_t i = 0; i < size; ++i) {
+            waveform[i] = static_cast<int8_t>(std::clamp<int>(dist(gen), -128, 127));
+        }
+        return waveform;
+    }
+
+
+
+// exponential
+
+    std::vector<int8_t> GenerateWaveForm_ExponentialGrowth(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            waveform[i] = static_cast<int8_t>(127 * (std::exp(t) - 1) / (std::exp(1) - 1));
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_ExponentialDecay(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            double value = std::exp(-5 * t); // Exponential decay
```

Comment:
Exponential decay can also just be computed by multiplying the previous value by a fixed fraction each iteration. Then you can avoid the `std:exp` call.